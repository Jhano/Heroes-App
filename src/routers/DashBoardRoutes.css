body {
    /* Ubicaci칩n de la imagen */
    background-image: url("../data/fondo/fondo.jpg");
    /* Para dejar la imagen de fondo centrada, vertical y
    horizontalmente */
    background-position: center center;
    /* Para que la imagen de fondo no se repita */
    background-repeat: no-repeat;
    /* La imagen se fija en la ventana de visualizaci칩n para que la altura de la imagen no supere a la del contenido */
    background-attachment: fixed;
    /* La imagen de fondo se reescala autom치ticamente con el cambio del ancho de ventana del navegador */
    background-size: cover;
    /* Se muestra un color de fondo mientras se est치 cargando la imagen
    de fondo o si hay problemas para cargarla */
    background-color: black;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    -ms-background-size: cover;
}