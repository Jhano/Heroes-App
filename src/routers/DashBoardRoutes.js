//Sistema de rutas principal
import React from 'react';
import { Col } from 'react-bootstrap';
import {
    Redirect,
    Route,
    Switch,
  } from "react-router-dom";

import DcScreen from '../components/dc/DcScreen';
import HeroScreen from '../components/heroes/HeroScreen';
import MarvelScreen from '../components/marvel/MarvelScreen';
import SearchScreen from '../components/search/SearchScreen';
import NavBar from '../components/ui/NavBar';

import './DashBoardRoutes.css';

const DashBoardRoutes = () => {
    return (
            <>
                
                <Col lg="12">
                    <NavBar/> 
                </Col> 
                <Col 
                    lg="12" 
                    className='fondo'
                > 
                    <Switch>
                        <Route exact path="/marvel" component={MarvelScreen}/>
                        <Route exact path="/hero/:heroeId" component={HeroScreen}/>
                        <Route exact path="/dc" component={DcScreen}/>
                        <Route exact path="/search" component={SearchScreen}/>

                        <Redirect to="/marvel"/>
                    </Switch>
                </Col>
               
            </>
        
     );
}
 
export default DashBoardRoutes;