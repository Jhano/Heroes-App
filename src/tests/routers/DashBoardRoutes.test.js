import React from 'react';
import '@testing-library/jest-dom'
import { mount } from 'enzyme';
import DashBoardRoutes from '../../routers/DashBoardRoutes';
import AuthContext from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';


describe('Pruebas en <DashBoardRoutes />', () => {

    const contextValue = {
        user: {
            name: 'Alejandro',
            logged: true,
        },
        dispatch: jest.fn()
    }

    test('El componente se muestra correctamente', () => {

             const wrapper = mount( 
                    <AuthContext.Provider value={contextValue}>
                        <MemoryRouter>
                            <DashBoardRoutes />
                        </MemoryRouter>
                    </AuthContext.Provider>
                 );
       
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').first().text().trim()).toBe('Alejandro');
    })
})