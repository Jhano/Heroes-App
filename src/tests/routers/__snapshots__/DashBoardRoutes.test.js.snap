// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashBoardRoutes /> El componente se muestra correctamente 1`] = `
Array [
  <div
    className="col-lg-12"
  >
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            aria-current="page"
            className="nav-link active"
            data-rb-event-key={null}
            disabled={false}
            href="/marvel"
            onClick={[Function]}
            style={Object {}}
          >
            Marvel
          </a>
          <a
            aria-current={null}
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current={null}
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/search"
            onClick={[Function]}
          >
            Buscar
          </a>
        </div>
        <div
          className="justify-content-end navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="text-info nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Alejandro
          </a>
          <a
            className="nav-link"
            data-rb-event-key={null}
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            Logout
          </a>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="fondo col-lg-12"
  >
    <div
      className="container"
    >
      <div
        className="row"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <h1
          style={
            Object {
              "color": "whitesmoke",
            }
          }
        >
          MarvelScreen
        </h1>
        <div
          className="container"
        >
          <div
            className="animate__animated animate__fadeIn card-columns"
          >
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-spider.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Spider Man
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  Amazing Fantasy #15
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-captain.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Captain America
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  Captain America Comics #1
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-iron.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Iron Man
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  Tales of Suspense #39
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-thor.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Thor
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  Journey into Myster #83
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-hulk.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Hulk
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  The Incredible Hulk #1
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-wolverine.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Wolverine
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  The Incredible Hulk #180
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-daredevil.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Daredevil
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  Daredevil #1
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-hawkeye.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Hawkeye
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  Tales of Suspense #57
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-cyclops.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Cyclops
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  X-Men #1
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                className="card-img-top"
                src="./assets/heroes/marvel-silver.jpg"
                style={
                  Object {
                    "height": "16rem",
                  }
                }
              />
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Silver Surfer
                </div>
                <hr />
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <div
                  className="card-subtitle h6"
                >
                   
                  The Fantastic Four #48
                   
                </div>
              </div>
              <div
                className="card-footer"
              >
                <a
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-info btn-block btn-lg"
                    disabled={false}
                    style={
                      Object {
                        "borderRadius": "2rem",
                      }
                    }
                    type="button"
                  >
                     M치s..
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
